triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/tsmigrationrequest/migrations/V10VersionUpdater.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.triplesgenerator.events.consumers.tsmigrationrequest
+package migrations
+
+import cats.data.EitherT
+import cats.effect.Async
+import cats.syntax.all._
+import cats.MonadThrow
+import com.typesafe.config.ConfigFactory
+import io.renku.graph.config.RenkuUrlLoader
+import io.renku.graph.model.RenkuUrl
+import io.renku.graph.model.versions.RenkuVersionPair
+import io.renku.triplesgenerator.config.VersionCompatibilityConfig
+import io.renku.triplesgenerator.events.consumers.ProcessingRecoverableError
+import io.renku.triplesstore.{MigrationsConnectionConfig, SparqlQueryTimeRecorder}
+import org.typelevel.log4cats.Logger
+import reprovisioning.RenkuVersionPairUpdater
+import tooling.{MigrationExecutionRegister, RecoverableErrorsRecovery, RegisteredMigration}
+
+private class V10VersionUpdater[F[_]: MonadThrow: Logger](
+    versionPair:             RenkuVersionPair,
+    renkuVersionPairUpdater: RenkuVersionPairUpdater[F],
+    executionRegister:       MigrationExecutionRegister[F],
+    recoveryStrategy:        RecoverableErrorsRecovery = RecoverableErrorsRecovery
+) extends RegisteredMigration[F](V10VersionUpdater.name, executionRegister, recoveryStrategy) {
+
+  import recoveryStrategy._
+
+  protected[migrations] override def migrate(): EitherT[F, ProcessingRecoverableError, Unit] = EitherT {
+    renkuVersionPairUpdater
+      .update(versionPair)
+      .map(_.asRight[ProcessingRecoverableError])
+      .recoverWith(maybeRecoverableError[F, Unit])
+  }
+}
+
+private[migrations] object V10VersionUpdater {
+
+  val name: Migration.Name = Migration.Name("V10 version Updater")
+
+  def apply[F[_]: Async: Logger: SparqlQueryTimeRecorder]: F[Migration[F]] = for {
+    implicit0(ru: RenkuUrl)    <- RenkuUrlLoader[F]()
+    compatibility              <- VersionCompatibilityConfig.fromConfigF[F](ConfigFactory.load())
+    migrationsConnectionConfig <- MigrationsConnectionConfig[F]()
+    executionRegister          <- MigrationExecutionRegister[F]
+  } yield new V10VersionUpdater(compatibility.asVersionPair,
```

Comment:
Not sure if I understand: the version pair to update comes from the config - so just in case it stays and later the config changes - would this affect this migration? I.e. could it then update to something else than V10?

Reply:
Right, I was thinking about that for a bit. Initially, I thought, let's just set the hardcoded v10 and cli v2.3.0. However, I realised that:
* This new migration will be run only once and straight after the v10 migration is done, which means it will insert the exact versions which were used during the processing.
* In a case when the v10 migration spreads over more than one cli versions (very unlikely multiple schema versions), the migration I added will insert the last versions that were used at the end of the v10 migration. That I think made me to the decision that it's probably better to read the versions from the config rather than hardcode them.
* In a case of a fresh environment (completely new one), this new migration doesn't not really matter as it will insert the same versions as the Re-provisioning migration would have just set.